criu/util.c

```diff
@@ -655,6 +655,63 @@ int cr_system_userns(int in, int out, int err, char *cmd, char *const argv[], un
 	return ret;
 }
 
+pid_t fork_and_ptrace_attach(int (*child_setup)(void))
+{
+	pid_t pid;
+	int sk_pair[2], sk;
+	char c = 0;
+
+	if (socketpair(PF_LOCAL, SOCK_SEQPACKET, 0, sk_pair)) {
+		pr_perror("socketpair");
+		return -1;
+	}
+
+	pid = fork();
+	if (pid < 0) {
+		pr_perror("fork");
+		return -1;
+	} else if (pid == 0) {
```

Comment:
`else` is not necessary here. `return -1` would prevent the execution of subsequent lines.

Reply:
I'm sorry, couldn't you clarify this? `else` seems required here because we need to execute one part of the code in the parent process and the second one in the child process.

Reply:
Sorry, I meant something like the following:
```C
pid = fork();
if (pid < 0) {
	pr_perror("fork");
	return -1;
}

if (pid == 0) {
...
}
```
This is just a code-style change.

Reply:
@rst0git thanks. Fixed!